src/doghouse/adapters/git/git_adapter.py

```
@@ -0,0 +1,40 @@
+                id="local-uncommitted",
+                type=BlockerType.LOCAL_UNCOMMITTED,
+                message="Local uncommitted changes detected",
+                severity=BlockerSeverity.WARNING
+            ))
+            
+        # Check for unpushed commits on the current branch
+        branch_res = subprocess.run(["git", "branch", "--show-current"], capture_output=True, text=True)
+        branch = branch_res.stdout.strip()
+        if branch:
+            # Check for commits that are in branch but not in its upstream
+            unpushed = subprocess.run(
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)